net/uring/io_uring_linux.go

```diff
@@ -602,3 +602,35 @@ func (t *TUN) Close() error {
 	}
 	return err2
 }
+
+var (
+	mu sync.Mutex
```

Comment:
Given that these probes are pretty cheap, I'd be tempted to use a sync.Once and populate the entire capabilities map. Then the function can return a nil capabilities map if probing is unsupported and a regular (re-usable) capabilities map otherwise.

Reply:
I'm still not returning a map, and instead just populating a singleton and indexing into it later. Hopefully that's good enough